logprep/processor/amides/processor.py

```diff
@@ -0,0 +1,212 @@
+    class AmidesMetrics(Processor.ProcessorMetrics):
+        """Track statistics specific for Amides instances."""
+
+        total_cmdlines: int = 0
+        """Total number of command-lines processed."""
+        new_results: int = 0
+        """Number of command-lines that triggered classification and rule attribution."""
+        cached_results: int = 0
+        """Number of command-lines that could be resolved from cache."""
+        num_cache_entries: int = 0
+        """Absolute number of current cache entries."""
+        cache_load: float = 0.0
+        """Relative cache load."""
+        mean_misuse_detection_time: float = 0.0
+        """Mean processing time of command-lines classified by the misuse detection model."""
+        _mean_misuse_detection_time_sample_counter: int = 0
+        mean_rule_attribution_time: float = 0.0
+        """Mean processing time of command-lines classified by the rule attribution model."""
+        _mean_rule_attribution_time_sample_counter: int = 0
+
+        def update_mean_misuse_detection_time(self, new_sample):
+            """Updates the mean processing time of the misuse detection."""
+            new_mean, new_sample_counter = calculate_new_average(
+                self.mean_misuse_detection_time,
+                new_sample,
+                self._mean_misuse_detection_time_sample_counter,
+            )
+            self.mean_misuse_detection_time = new_mean
+            self._mean_misuse_detection_time_sample_counter = new_sample_counter
+
+        def update_mean_rule_attribution_time(self, new_sample):
+            """Updates the mean processing time of the rule attribution model."""
+            new_mean, new_sample_counter = calculate_new_average(
+                self.mean_rule_attribution_time,
+                new_sample,
+                self._mean_rule_attribution_time_sample_counter,
+            )
+            self.mean_rule_attribution_time = new_mean
+            self._mean_rule_attribution_time_sample_counter = new_sample_counter
+
+    __slots__ = (
+        "_misuse_detector",
+        "_rule_attributor",
+        "_normalizer",
+        "_evaluate_cmdline_cached",
+    )
+
+    _misuse_detector: MisuseDetector
+    _rule_attributor: RuleAttributor
+    _normalizer: CommandLineNormalizer
+    _evaluate_cmdline_cached: _lru_cache_wrapper
+
+    rule_class = AmidesRule
+
+    def __init__(self, name: str, configuration: Processor.Config, logger: Logger):
+        super().__init__(name=name, configuration=configuration, logger=logger)
+        self.metrics = self.AmidesMetrics(
+            labels=self.metric_labels,
+            generic_rule_tree=self._generic_tree.metrics,
+            specific_rule_tree=self._specific_tree.metrics,
+        )
+        self._normalizer = CommandLineNormalizer(max_num_values_length=3, max_str_length=30)
+        self._evaluate_cmdline_cached = lru_cache(maxsize=self._config.max_cache_entries)(
+            self._evaluate_cmdline
+        )
```

Comment:
There is a good chance, to eliminate the `__init__` completly, by renaming the Metrics as mentioned above and making the `self._nomalizer` and `self._evaluate_cmdline_cached` a `@cached_property` returning the same things

Reply:
I made `self._normalizer` and `self._evaluate_cmdline_cached` a cached property. As `AmidesMetrics` is now `ProcessorMetrics` (and behaviour looked good so far), I could remove the `__init__` completetly within the latest changes.